metaflow/plugins/kfp/kfp.py

```diff
@@ -105,6 +106,61 @@ def _get_retries(node):
 
         return max_user_code_retries, max_user_code_retries + max_error_retries
 
+    @staticmethod
+    def _get_resource_requirements(node):
+        """
+        Get resource request or limit for a Metaflow step (node).
+        If @resources is also present, the maximum value is used for resource request, and lowest
+        value is used for resource limit.
+
+        Eventually resource request and limits link back to kubernetes, see
+        https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+
+        Supported parameters: 'cpu', 'cpu_limit', 'gpu', 'gpu_vendor', 'memory', 'memory_limit'
+        Keys with no suffix set resource request (minimum);
+        keys with 'limit' suffix sets resource limit (maximum).
+
+        Default unit for memory is megabyte.
+
+        Example using resource decorator:
+            @resource(cpu=0.5, cpu_limit=2, gpu=1, memory=300)
+            @step
+            def my_kfp_step(): ...
+        """
+
+        resource = dict()
+        for deco in node.decorators:
+            if isinstance(deco, ResourcesDecorator):
+                for k, v in deco.attributes.items():
```

Comment:
Can we please try to get away from single letter variable names, can you find some better names for these.